stan/math/prim/fun/grad_pFq.hpp

```diff
@@ -2,344 +2,201 @@
+ * Noting the the recurrence relation for the digamma function:
+ * \f$ \psi(x + 1) = \psi(x) + \frac{1}{x} \f$, the gradients for the
+ * function w.r.t a & b then simplify to:
+ * \f$
+ * \frac{\partial }{\partial a_1} =
+ *  \sum_{k=1}^{\infty}{
+ *    \frac
+ *      {\left(1 + \sum_{m=0}^{k-1}\frac{1}{m+a_1}\right)
+ *        * (\prod_{j=1}^p\left(a_j\right)_k\right)z^k}
+ *      {k!\prod_{j=1}^q\left(b_j\right)_k}}
+ *    - _pF_q(a_1,...,a_p;b_1,...,b_q;z)
+ * \f$
+ * \f$
+ * \frac{\partial }{\partial b_1} =
+ *  _pF_q(a_1,...,a_p;b_1,...,b_q;z) -
+ *  \sum_{k=1}^{\infty}{
+ *    \frac
+ *      {\left(1 + \sum_{m=0}^{k-1}\frac{1}{m+b_1}\right)
+ *        * \left(\prod_{j=1}^p\left(a_j\right)_k\right)z^k}
+ *      {k!\prod_{j=1}^q\left(b_j\right)_k}}
+ * \f$
  *
- * @tparam calc_a Boolean for whether to calculate derivatives wrt to 'a'
- * @tparam calc_b Boolean for whether to calculate derivatives wrt to 'b'
- * @tparam calc_z Boolean for whether to calculate derivatives wrt to 'z'
- * @tparam TupleT Type of tuple containing objects to evaluate gradients into
+ * @tparam CalcA Boolean for whether to calculate derivatives wrt to 'a'
+ * @tparam CalcB Boolean for whether to calculate derivatives wrt to 'b'
+ * @tparam CalcZ Boolean for whether to calculate derivatives wrt to 'z'
+ * @tparam TpFq Scalar type of hypergeometric_pFq return value
  * @tparam Ta Eigen type with either one row or column at compile time
  * @tparam Tb Eigen type with either one row or column at compile time
  * @tparam Tz Scalar type
- * @param[in] grad_tuple Tuple of references to evaluate gradients into
+ * @param[in] pfq_val Return value from the hypergeometric_pFq function
  * @param[in] a Vector of 'a' arguments to function
  * @param[in] b Vector of 'b' arguments to function
  * @param[in] z Scalar z argument
  * @param[in] precision Convergence criteria for infinite sum
- * @param[in] outer_steps Maximum number of iterations for infinite sum
- * @param[in] inner_steps Maximum number of iterations for infinite sum
- * @return Generalized hypergeometric function
+ * @param[in] max_steps Maximum number of iterations
+ * @return Tuple of gradients
  */
-template <bool calc_a, bool calc_b, bool calc_z, typename TupleT, typename Ta,
-          typename Tb, typename Tz,
-          require_all_eigen_vector_t<Ta, Tb>* = nullptr,
-          require_stan_scalar_t<Tz>* = nullptr>
-void grad_pFq_impl(TupleT&& grad_tuple, const Ta& a, const Tb& b, const Tz& z,
-                   double precision, int outer_steps, int inner_steps) {
+template <bool CalcA = true, bool CalcB = true, bool CalcZ = true,
+          typename TpFq, typename Ta, typename Tb, typename Tz,
+          typename T_Rtn = return_type_t<Ta, Tb, Tz>,
+          typename Ta_Rtn = promote_scalar_t<T_Rtn, plain_type_t<Ta>>,
+          typename Tb_Rtn = promote_scalar_t<T_Rtn, plain_type_t<Tb>>>
+std::tuple<Ta_Rtn, Tb_Rtn, T_Rtn> grad_pFq(const TpFq& pfq_val, const Ta& a,
```

Comment:
Questions:
1. do we intend for `grad_pFq()` to be exposed to users through the Stan language? (If so, we should change the template signature here to match what would be appropriate to expose)
2. Is there a reason this was moved out of the `stan::math::internal` namespace to `stan::math`?
3. If I'm looking at the ways this is called, am I correct in thinking that this is primitive-only? (I could be incorrect about that; maybe the fvar path calls it with the inner type.)

Reply:
> 1. do we intend for grad_pFq() to be exposed to users through the Stan language? (If so, we should change the template signature here to match what would be appropriate to expose)

Nope, this is only for computing gradients for the `hypergeometric_` functions, not for end-users. It's currently used for the `hypergeometric_2F2`, `hypergeometric_3F2`, and `hypergeometric_pFq` functions, and will be used for [Boost's Hypergeometric functions](https://www.boost.org/doc/libs/1_84_0/libs/math/doc/html/math_toolkit/hypergeometric.html) 

> 2. Is there a reason this was moved out of the stan::math::internal namespace to stan::math?

Previously only an `_impl` component (`grad_pFq_impl`) was the in the internal namespace, `grad_pFq` was always in the `stan::math::` namespace

> 3. If I'm looking at the ways this is called, am I correct in thinking that this is primitive-only? (I could be incorrect about that; maybe the fvar path calls it with the inner type.)

This is also for `fvar`/higher-order inputs as well (tested as part of the new `mix` tests for `hypergeometric_pFq` in this PR)